pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -880,8 +878,6 @@ impl DeltaLayerInner {
         self.do_reads_and_update_state(reads, reconstruct_state, ctx)
             .await;
 
-        reconstruct_state.on_lsn_advanced(&keyspace, lsn_range.start);
```

Comment:
Am I understanding it correctly that we can no longer apply this optimization because we're dispatching all the IOs concurrently and we can't know which delta layer reads can be omitted until we've read the image layers?

I guess that's reasonable, unless we want to be clever and use read barriers at each image layer or some other heuristic (not clear that this would be a net win anyway). But what about the default `Serial` path, will that now see a performance regression because we removed this?

Reply:
We used to store page images in the pageserver page cache when the vectored read path was implemented. Hence, I also implemented support for using the cache on the vectored read path (that's what this function was part of). Nowadays, we no longer store page images in the page cache (only laryer indices are cached in-mem).

TLDR: I don't expect any perf diff since this wasn't used.

Reply:
Got it, thanks for the background.